api/src/main/java/org/openmrs/module/oauth2login/OAuth2LoginActivator.java

```diff
@@ -27,6 +34,31 @@ public class OAuth2LoginActivator extends BaseModuleActivator implements DaemonT
 	
 	private DaemonToken daemonToken;
 	
+	/**
+	 * @see #willStart()
+	 */
+	@Override
+	public void willStart() {
+		// Checks if OAuth2 is enabled, if not, stops the module and unloads it
+		try {
+			Properties oauth2Props = PropertyUtils.getOAuth2Properties();
+			boolean moduleEnabled = Boolean.parseBoolean(oauth2Props.getProperty(OAUTH2_ENABLED_PROPERTY, "true"));
+			if (!moduleEnabled) {
+				log.info("OAuth2 is disabled. Skipping module start.");
+				// Stop the module if it is already started
+				if (ModuleFactory.isModuleStarted(MODULE_ARTIFACT_ID)) {
```

Comment:
Did we say we want to stop the module from starting? Or stop it from registering the auth scheme?

Reply:
IIRC, Romain mentioned the best approach was to stop the module from starting at all. But we weren't sure if that was possible, so we agreed that just stopping it from registering the auth scheme would be enough. IMO, this is better.